File: app/src/main/java/com/ireddragonicy/konabessnext/core/strategy/SingleBinStrategy.java
package com.ireddragonicy.konabessnext.core.strategy;

import java.util.ArrayList;
import java.util.List;
import com.ireddragonicy.konabessnext.model.Bin;

public class SingleBinStrategy extends BaseChipArchitecture {

    @Override
    public boolean isStartLine(String line) {
        return line.equals("qcom,gpu-pwrlevels {");
    }

    @Override
    public List<String> generateTable(List<Bin> bins) {
        List<String> lines = new ArrayList<>();
        if (!bins.isEmpty()) {
            lines.add("qcom,gpu-pwrlevels {"); // Single bin header is constant
            generateBinContent(bins.get(0), lines);
        }
        return lines;
    }
}

File: app/src/main/java/com/ireddragonicy/konabessnext/core/strategy/SingleBinStrategy.kt
package com.ireddragonicy.konabessnext.core.strategy

import com.ireddragonicy.konabessnext.model.Bin
import java.util.ArrayList

class SingleBinStrategy : BaseChipArchitecture() {

    override fun isStartLine(line: String): Boolean {
        return line == "qcom,gpu-pwrlevels {"
    }

    override fun generateTable(bins: List<Bin>): List<String> {
        val lines = ArrayList<String>()
        if (bins.isNotEmpty()) {
            lines.add("qcom,gpu-pwrlevels {") // Single bin header is constant
            generateBinContent(bins[0], lines)
        }
        return lines
    }
}
